CONTRIBUTING.md

```diff
@@ -0,0 +1,72 @@
+- When posting Python stack traces, please quote them using
+[Markdown blocks](https://help.github.com/articles/creating-and-highlighting-code-blocks/).
+- Tag the issue with `bug`
+- Remember that this is a volunteer-driven project, and that contributions are welcome :)
+
+**SLO**
+
+- We will respond to all new issues within 24 hours
+- For any serious (production breaking) bug we will try to resolve ASAP and do a hotfix release
+- For other bugs we will try to resolve them within the next 2 releases (There is a release every 2 weeks).
+
+### Submit Ideas or Feature Requests
+
+The best way is to file an issue on GitHub:
+
+- Explain in detail how it would work.
+- Keep the scope as narrow as possible, to make it easier to implement.
+- Tag the issue with `feature-request`
+
+### Improve Documentation
+
+Opta could always use better documentation, so feel free to open a Pull Request in our [docs repo](https://github.com/run-x/opta-docs)
+
+### Ask Questions
+
+Please come and hangout in our [Slack](https://slack.opta.dev/).
+
+## Pull Request Guidelines
+
+A philosophy we would like to strongly encourage is
+
+> Before creating a PR, create an issue.
+
+The purpose is to separate problem from possible solutions.
+
+**Bug fixes:** If you’re only fixing a small bug, it’s fine to submit a pull request right away but we highly recommend to file an issue detailing what you’re fixing. This is helpful in case we don’t accept that specific fix but want to keep track of the issue. Please keep in mind that the project maintainers reserve the rights to accept or reject incoming PRs, so it is better to separate the issue and the code to fix it from each other. In some cases, project maintainers may request you to create a separate issue from PR before proceeding.
+
+**Feature/Large changes:** If you intend to change the public API, or make any non-trivial changes to the implementation, we require you to file a new issue. This lets us reach an agreement on your proposal before you put significant effort into it. You are welcome to submit a PR along with the issue (sometimes necessary for demonstration), but we will not review/merge the code until there is an agreement on the issue.
+
+In general, small PRs are always easier to review than large PRs. The best practice is to break your work into smaller independent PRs and refer to the same issue. This will greatly reduce turnaround time.
+
+If you wish to share your work which is not ready to merge yet, create a [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/). This will enable maintainers and the CI runner to prioritize mature PR's.
+
+Finally, never submit a PR that will put the main branch in broken state. If the PR is part of multiple PRs to complete a large feature and cannot work on its own, you can create a feature branch and merge all related PRs into the feature branch before creating a PR from feature branch to main.
```

Comment:
Contributors may not know how to test this - whether their PR is breaking the main branch. We should point them to which tests they need to perform before submitting PR (e.g. unit tests, linting with scripts/lint.py etc) , and then which ones will be performed by the core team later (e.g. CircleCI).

Reply:
So we already have a setting that if unit tests etc are failing the PR cannot be merged. I think it's fine to just have this line as a guideline. If someone still sends a PR with broken tests, we can remind them of this. They can't merge it anyway.